crates/generated_parser/src/early_errors.rs

```diff
@@ -909,6 +971,80 @@ impl VarEarlyErrorsContext for LexicalForBodyEarlyErrorsContext {
     }
 }
 
+// ===========================================================================
+// LabelledStatements
+// https://tc39.es/ecma262/#sec-labelled-statements
+// ===========================================================================
+
+pub struct LabelledStatementEarlyErrorsContext {
+    name: SourceAtomSetIndex,
+    is_loop: bool,
+}
+
+impl LabelledStatementEarlyErrorsContext {
+    pub fn new(name: SourceAtomSetIndex, is_loop: bool) -> Self {
+        Self { name, is_loop }
+    }
+
+    pub fn check_duplicate_label<'alloc>(
+        &self,
+        name: SourceAtomSetIndex,
```

Comment:
how about naming this `inner_label` or something? so that it's clear what this is comparing against.

there can be 2 ways to detect duplicate labels.
compare against all ancestor labels, or compare against all inner labels.
and this takes latter, right?

Reply:
yes, this takes the latter. We don't have enough information to do the former without the context stack, as there is no distinction between LabelIdentifiers that are part of LabelledStatements and LabelIdentifiers that are part of control statements.